libraries/abstractions/common_io/test/test_iot_tsensor.c

```diff
@@ -61,6 +63,17 @@ TEST_GROUP( TEST_IOT_TSENSOR );
  */
 TEST_SETUP( TEST_IOT_TSENSOR )
 {
+    if( xtestIotTsensorMinSemaphore == NULL )
```

Comment:
It think the test should check this and assert and fail if it already exists. What will happen if a prior test hands you a locked semaphore?

Reply:
There are a couple of these further down also, this applies to all of them.

Reply:
I fixed them.